app/code/Magento/Checkout/Api/PaymentInformationManagementInterface.php

```diff
@@ -11,6 +11,21 @@
  */
 interface PaymentInformationManagementInterface
 {
+    /**
+     * Set payment information and place order for a specified cart.
+     *
+     * @param int $cartId
+     * @param \Magento\Quote\Api\Data\PaymentInterface $paymentMethod
+     * @param \Magento\Quote\Api\Data\AddressInterface|null $billingAddress
+     * @throws \Magento\Framework\Exception\CouldNotSaveException
+     * @return \Magento\Checkout\Api\Data\PlaceOrderDetailsInterface
+     */
+    public function placeOrder(
```

Comment:
Hi @gabrieldagama 

I discovered that at order place action the check on coupon usage limit is made after order place on this service \Magento\Sales\Model\Service\OrderService::place. And this ends up in the current situation where the place order process could not be stopped in case of limit has been passed. 
see: app/code/Magento/SalesRule/etc/di.xml:192

Also according to the Bug description
> Expected result (*)
- Only one order should be successful.
- Second order should result in an error due to the coupon code or be placed without the discount.

Personally I find unacceptable the second option "be placed without the discount." because the customer should always know the order total. 

So we need to add a new way of placing order where error could be shown to customer. 
The original order placing method you have mentioned savePaymentInformationAndPlaceOrder is returning an order id in case of success, but in case of an error during the order place process only a general error message is shown to user. In this case the customer could abandon the process since is not aware of the problem. 


So answering to your questions, with this new placeOrder method I tried to follow backward compatible rules, because original method is still called, and the new method is only changing the return type for the API call in order to be able to display a specific error in front-end.